File: 351001/polozkov/src/main/java/org/polozkov/entity/issue/Issue.java
package org.polozkov.entity.issue;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.polozkov.entity.comment.Comment;
import org.polozkov.entity.label.Label;
import org.polozkov.entity.user.User;

import java.time.LocalDateTime;
import java.util.List;

@Getter
@Setter
@Entity
public class Issue {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;

    private String content;

    private LocalDateTime created;

    private LocalDateTime modified;

    @ManyToOne()
    @JoinColumn(name = "user_id")
    private User user;

    @OneToMany(mappedBy = "issue", fetch = FetchType.LAZY)
    private List<Comment> comments;

    @ManyToMany(mappedBy = "issues", fetch = FetchType.LAZY)
    private List<Label> labels;

}

File: 351002/Yanachkin/Lab1-5/lab1/Models/Entities/Issue.cs
namespace lab1.Models.Entities;

public class Issue : BaseEntity
{
    public string Title { get; set; } = null!;
    public string Content { get; set; } = null!;

    public long EditorId { get; set; }
    public Editor Editor { get; set; } = null!;

    public DateTime Created { get; set; }
    public DateTime Modified { get; set; }

    public ICollection<Label> Labels { get; set; } = new List<Label>();
}
